lib/puppet/provider/package/te_agent_bin.rb

```diff
@@ -14,16 +14,41 @@
   has_feature :versionable
   has_feature :install_options
 
+  confine :osfamily => [ :RedHat ]
```

Comment:
This provider is used for all non-Windows operating systems.

Reply:
Removed confines